source/VirtualDesktop/VirtualDesktop.static.cs

```diff
@@ -139,19 +76,40 @@ public static VirtualDesktop FromHwnd(IntPtr hwnd)
 
 			if (hwnd == IntPtr.Zero) return null;
 
-			IVirtualDesktop desktop;
 			try
 			{
-				var desktopId = ComObjects.VirtualDesktopManager.GetWindowDesktopId(hwnd);
-				desktop = ComObjects.VirtualDesktopManagerInternal.FindDesktop(ref desktopId);
+				var desktopId = ComInterface.VirtualDesktopManager.GetWindowDesktopId(hwnd);
+				return ComInterface.VirtualDesktopManagerInternal.FindDesktop(ref desktopId);
 			}
 			catch (COMException ex) when (ex.Match(HResult.REGDB_E_CLASSNOTREG, HResult.TYPE_E_ELEMENTNOTFOUND))
 			{
 				return null;
 			}
-			var wrapper = _wrappers.GetOrAdd(desktop.GetID(), _ => new VirtualDesktop(desktop));
+		}
+
+		internal static bool GetIsSupported()
+		{
+			return _isSupported ?? (_isSupported = Core()).Value;

```

Comment:
`_isSupported` defaults to `true` so when when my application calls this for the first time `Core()` is never called and `ProviderInternal` never Initialized.

The sample applications work fine though.